pkg/cloud/packet/scope/machine.go

```diff
@@ -208,3 +237,160 @@ func (m *MachineScope) GetRawBootstrapData() ([]byte, error) {
 
 	return value, nil
 }
+
+func getProviderIDPrefix(ctx context.Context, mgmtClient client.Client, workloadClientGetter remote.ClusterClientGetter, cluster *clusterv1.Cluster, machine *clusterv1.Machine, packetMachine *infrav1.PacketMachine) (string, error) {
```

Comment:
I could use an explanation comment of how and why providerIDFromCloudProviderDeployments is expected to work.

Reply:
Added some additional comments around the rationale and expected behavior of the functions